src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UserDefinedTypeDescriptor.cs

```diff
@@ -732,11 +732,11 @@ private uint GetClassTypeIndex(TypeDesc type, bool needsCompleteType)
                 statics[i] = staticsDescs[i];
             }
 
-            LayoutInt elementSize = defType.GetElementSize();
-            int elementSizeEmit = elementSize.IsIndeterminate ? 0xBAAD : elementSize.AsInt;
+            LayoutInt instanceSize = defType.InstanceByteCount;
```

Comment:
I'd not use number of bytes on the GC heap as the size for value types:

```suggestion
            LayoutInt instanceSize = defType.IsValueType ? defType.InstanceFieldSize : defType.InstanceByteCount;
```